### PR TITLE
C/C++: tainting: Track `this->xyz` l-values

### PR DESCRIPTION
**Pull request by ihji**
Test Plan: https://github.com/returntocorp/semgrep-proprietary/pull/1052

### CODE REVIEW

File: src/tainting/Taint_lval_env.ml [hunk 1]
```diff
@@ -79,6 +79,8 @@ let normalize_lval lval =
   let* base, rev_offset =
```
Comment (IagoAbal) [inline src/tainting/Taint_lval_env.ml:79]: There is no changelog.

**Review by IagoAbal — APPROVED** [review]

File: src/tainting/Taint_lval_env.ml [hunk 2]
```diff
@@ -79,6 +79,8 @@ let normalize_lval lval =
...
     match base with
     | Var _ -> Some (base, rev_offset)
+    (* explicit dereference of `this` e.g. `this->x` *)
```
Comment (IagoAbal) [inline src/tainting/Taint_lval_env.ml:85]: You should be able to add an intra-procedural test here. Something like this:

```cpp
ptr->a = "tainted"
ptr->b = "safe"
// ruleid:
sink(ptr->a); 
// ok:
sink(ptr->b);
```

File: src/tainting/Taint_lval_env.ml [hunk 3]
```diff
@@ -79,6 +79,8 @@ let normalize_lval lval =
...
+    | Mem { e = Fetch { base = VarSpecial (This, _); rev_offset = [] }; _ }
```
Comment (IagoAbal) [inline src/tainting/Taint_lval_env.ml:86]: Shouldn't we handle more than just `this` here? Then we can write tests for `ptr->a` etc as suggested in a previous comment.

File: src/tainting/Taint_lval_env.ml (hunk 1 above)
Comment (ihji) [reply src/tainting/Taint_lval_env.ml:79]: Added

File: src/tainting/Taint_lval_env.ml (hunk 2 above)
Comment (ihji) [reply src/tainting/Taint_lval_env.ml:85]: Added

File: src/tainting/Taint_lval_env.ml (hunk 3 above)
Comment (ihji) [reply src/tainting/Taint_lval_env.ml:86]: Done

File: changelog.d/gh-1058.fixed [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  obj->b = SAFE_STR;
+
+  // ok: cpp-tainted-field-ptr
+  sink(obj->b, __LINE__);
+  // ruleid: cpp-tainted-field-ptr
+  sink(obj->a, __LINE__);
+}
+```
+
+This can be matched by the rule:
```
Comment (IagoAbal) [inline changelog.d/gh-1058.fixed:19]: In these toy examples for illustration purposes I think it's fine to skip the rule, it doesn't add much value but makes the changelog a lot longer. If we want to provide something for the user to reproduce, I think it's better a Playground permalink.

**Pull request merged**